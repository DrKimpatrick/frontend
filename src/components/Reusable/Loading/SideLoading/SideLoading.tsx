import React from 'react';
import { ClipLoader } from 'react-spinners';
import { css } from '@emotion/core';
import './SideLoading.scss';

const override = css`
  display: block;
`;

interface Props {
  size?: number;
}
const SideLoading = (props: Props) => {
  const { size } = props;

  return (
    <div className="sideLoading">
      {size ? (
        <ClipLoader color="#010a26" css={override} size={size} />
      ) : (
        <ClipLoader color="#010a26" css={override} />
      )}
    </div>
  );
};

export default SideLoading;
