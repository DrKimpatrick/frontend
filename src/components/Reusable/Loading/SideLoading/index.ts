import SideLoading from './SideLoading';

export { SideLoading };
