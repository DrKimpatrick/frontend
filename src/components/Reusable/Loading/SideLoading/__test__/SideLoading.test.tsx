import React from 'react';
import SideLoading from '../SideLoading';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
describe('Side Loading', () => {
  afterEach(cleanup);
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<SideLoading />, div);
  });

  it('should create a snapshot', () => {
    const tree = renderer.create(<SideLoading />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render loading with provided size', () => {
    const tree = renderer.create(<SideLoading size={20} />).toJSON();

    expect(tree).toBeTruthy();
  });
});
