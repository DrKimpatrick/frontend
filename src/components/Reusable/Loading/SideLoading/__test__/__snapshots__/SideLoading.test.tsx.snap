// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Loading should create a snapshot 1`] = `
<div
  className="sideLoading"
>
  <div
    className="css-6fk06n"
  />
</div>
`;
